### PR TITLE
nhttp: Return json errors when asked for

### PR DESCRIPTION
**Pull request by vorner**
If the client asks for application/json (the PrusaLink pages), comply
and supply the errors in that format.

### CODE REVIEW

File: tests/unit/lib/WUI/automata/generated.cpp [hunk 1]
```diff
@@ -221,6 +221,14 @@ TEST_CASE("Connection keep alive") {
     REQUIRE_FALSE(ex.contains_enter(Names::ConnectionClose));
 }
 
+TEST_CASE("Accep json") {
```
Comment (espr14) [inline tests/unit/lib/WUI/automata/generated.cpp:224]:
```suggestion
TEST_CASE("Accept json") {
```

File: lib/WUI/nhttp/status_page.cpp [hunk 2]
```diff
@@ -20,10 +21,19 @@ Step StatusPage::step(std::string_view, bool, uint8_t *output, size_t output_siz
     const StatusText &text = StatusText::find(status);
 
     char content_buffer[128];
+    ContentType ct;
```
Comment (espr14) [inline lib/WUI/nhttp/status_page.cpp:24]:
```suggestion
    ContentType ct = ContentType::TextPlain;
```

File: lib/WUI/nhttp/status_page.cpp [hunk 3]
```diff
@@ -20,10 +21,19 @@ Step StatusPage::step(std::string_view, bool, uint8_t *output, size_t output_siz
...
+        ct = ContentType::TextPlain;
+    } else if (json_content) {
+        const char *title = text.text;
+        JSONIFY_STR(title);
+        JSONIFY_STR(extra_content);
+        snprintf(content_buffer, sizeof(content_buffer), "{\"title\": \"%u: %s\",\"message\":\"%s\"}", static_cast<unsigned>(status), title_escaped, extra_content_escaped);
+        ct = ContentType::ApplicationJson;
     } else {
         snprintf(content_buffer, sizeof(content_buffer), "%u: %s\n\n%s\n", static_cast<unsigned>(status), text.text, extra_content);
+        ct = ContentType::TextPlain;
```
Comment (espr14) [inline lib/WUI/nhttp/status_page.cpp:36]:
```suggestion
```

File: lib/WUI/nhttp/status_page.cpp [hunk 4]
```diff
@@ -20,10 +21,19 @@ Step StatusPage::step(std::string_view, bool, uint8_t *output, size_t output_siz
     const StatusText &text = StatusText::find(status);
 
     char content_buffer[128];
+    ContentType ct;
     if (status == Status::NoContent || status == Status::NotModified) {
         content_buffer[0] = 0;
+        ct = ContentType::TextPlain;
```
Comment (espr14) [inline lib/WUI/nhttp/status_page.cpp:27]:
```suggestion
```

File: utils/gen-automata/http.py [hunk 5]
```diff
@@ -144,48 +144,71 @@ def read_boundary():
     return auto, end, False
 
 
-def connection_header():
+def keywoarded_header(keywords):
```
Comment (espr14) [inline utils/gen-automata/http.py:147]:
```suggestion
def keyworded_header(keywords):
```

File: utils/gen-automata/http.py [hunk 6]
```diff
@@ -144,48 +144,71 @@ def read_boundary():
...
+        'keep-alive': 'ConnectionKeepAlive',
+    })
+
+
+def accept_header():
+    """
+    Looking at the accept header.
+
+    We are cheating here a bit. We only want to know if the other side wants a
+    applicatio/json error messages and that doesn't list bunch of different
```
Comment (espr14) [inline utils/gen-automata/http.py:204]:
```suggestion
    application/json error messages and that doesn't list bunch of different
```

File: utils/gen-automata/http.py [hunk 7]
```diff
@@ -144,48 +144,71 @@ def read_boundary():
...
+    ones. We pick the first one and check it is the one we want.
+    """
+    return keywoarded_header({
```
Comment (espr14) [inline utils/gen-automata/http.py:207]:
```suggestion
    return keyworded_header({
```

**Review by espr14 — CHANGES_REQUESTED** [review]

File: lib/WUI/nhttp/status_page.cpp (hunk 2 above)
Comment (vorner) [reply lib/WUI/nhttp/status_page.cpp]: I can understand why you don't want an unitialized variable around, so I'm giving it some initial value now (corresponding to "unknown"). But I believing having one branch of the if special and defaulting the value there by not setting it hurts readability. See the current version.

**Review by espr14 — APPROVED** [review]

**Pull request merged**